Slm customized 1 v1.6.0 psa tls ipv6

pirun opened the pull request:
IPV6 feature based on PSA TLS branch.

lats1980 commented on applications/serial_lte_modem/doc/TCPIP_AT_commands.rst:
```diff
@@ -1097,7 +1110,10 @@ Response syntax
 
 ::
+   .. option:: CONFIG_SLM_HTTPC - HTTP client support in SLM .. is enabled.
+   #XTCPSVR: <listen_socket_handle>,<income_socket_handle>,<data_mode>,<timeout>,<data_mode>,<family>
```
Redundant option?

pirun replied:
**_.. option:: CONFIG_SLM_HTTPC - HTTP client support in SLM .. is disabled._** should be removed.
It might be something wrong when I port ipv6.

lats1980 commented on applications/serial_lte_modem/doc/TCPIP_AT_commands.rst:
```diff
@@ -1016,21 +1022,28 @@ Syntax
 
-   #XTCPSVR=<op>[<port>[,<sec_tag>]]
+   .. option:: CONFIG_SLM_HTTPC - HTTP client support in SLM .. is disabled.
+   #XTCPSVR=<op>[,<port>[,<sec_tag>]]
```
Redundant line?
.. option:: CONFIG_SLM_HTTPC - HTTP client support in SLM .. is disabled.

pirun replied:
**_.. option:: CONFIG_SLM_HTTPC - HTTP client support in SLM .. is disabled._** should be removed.
It might be something wrong when I port ipv6.

lats1980 commented on applications/serial_lte_modem/src/slm_at_tcp_proxy.c:
```diff
@@ -155,16 +161,17 @@ static int do_tcp_server_start(uint16_t port)
+		proxy.sock = socket(proxy.family, SOCK_STREAM, IPPROTO_TCP);
 	} else {
-		proxy.sock = socket(AF_INET, SOCK_STREAM | SOCK_NATIVE, IPPROTO_TLS_1_2);
+		proxy.sock = socket(proxy.family, SOCK_STREAM, IPPROTO_TLS_1_2);
```
Please keep SOCK_STREAM | SOCK_NATIVE in order to use native TLS for server

lats1980 commented on applications/serial_lte_modem/src/slm_at_tcp_proxy.c:
```diff
@@ -155,16 +161,17 @@ static int do_tcp_server_start(uint16_t port)
 	/* Open socket */
 	if (proxy.sec_tag == INVALID_SEC_TAG) {
-		proxy.sock = socket(AF_INET, SOCK_STREAM | SOCK_NATIVE, IPPROTO_TCP);
+		proxy.sock = socket(proxy.family, SOCK_STREAM, IPPROTO_TCP);
```
SOCK_STREAM is fine. Original flag SOCK_STREAM | SOCK_NATIVE is not correct.

pirun committed:
applications: serial_lte_modem: IPv6 feature backport.

IPV6 features backport.

Reference
https://github.com/nrfconnect/sdk-nrf/commit/
9073abd818e9942128cd08ce53bee47209e64202

https://github.com/nrfconnect/sdk-nrf/commit/
a0fc38d842f4772c04f957b2d473644b567a1950

https://github.com/nrfconnect/sdk-nrf/commit/
8db4cdbd28984cfca8e1790fd6cb0d783ad8ef4f

https://github.com/nrfconnect/sdk-nrf/commit/
fa6fad0c3997530386fbe7908113a8d638e6596f

https://github.com/nrfconnect/sdk-nrf/commit/
1e3b130505bdb6b0690be0fdcadb41a49b91ca40

https://github.com/nrfconnect/sdk-nrf/commit/
c45d13fabec669f7b3b32bf22e754b598f2bce77

https://github.com/nrfconnect/sdk-nrf/commit/
8711032238e147725630fd3a8a45fe96ebaacdfa

https://github.com/nrfconnect/sdk-nrf/commit/
3eef36295b144869fdbd099b3dc54488fa129f10

https://github.com/nrfconnect/sdk-nrf/commit/
7ded0c823bc83114569cf539f95b7974138e4fd5

https://github.com/nrfconnect/sdk-nrf/commit/
b62dc396e51f1da0cad82a141ef891519d264923

Fix AT#XRECV and AT#XRECVFROM

If <length> of these commands exceeds CONFIG_SLM_SOCKET_RX_MAX,
rx_data will be overflow and corrupt slm_operation_mode.
Then there is no handler to process at command.

FIX:TLS SOCKET can't use IP.

Signed-off-by: Pirun Lee <pirun.lee@nordicsemi.no>
